internal/resolver/delegatingresolver/delegatingresolver_ext_test.go

```diff
@@ -0,0 +1,480 @@
+		resolvedTargetTestAddr1 = "1.2.3.4:8080"
+		resolvedTargetTestAddr2 = "1.2.3.5:8080"
+	)
+	targetResolver := manual.NewBuilderWithScheme("test") // Set up a manual resolver to control the address resolution.
+	target := targetResolver.Scheme() + ":///" + targetTestAddr
+
+	// Create a delegating resolver with no proxy configuration
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	if _, err := delegatingresolver.New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, targetResolver, false); err != nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	// Update the manual resolver with a test address.
+	targetResolver.UpdateState(resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedTargetTestAddr1}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	// Verify that the delegating resolver outputs the same addresses, as returned
+	// by the target resolver.
+	wantState := resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedTargetTestAddr1}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	}
+
+	var gotState resolver.State
+	select {
+	case gotState = <-stateCh:
+	case <-time.After(defaultTestTimeout):
+		t.Fatal("Timeout when waiting for a state update from the delegating resolver")
+	}
+
+	if diff := cmp.Diff(gotState, wantState); diff != "" {
+		t.Fatalf("Unexpected state from delegating resolver. Diff (-got +want):\n%v\n got state : %s, want state %s\n", diff, pretty.ToJSON(gotState), pretty.ToJSON(wantState))
+	}
+}
+
+// setupDNS registers a new manual resolver for the DNS scheme, effectively
+// overwriting the previously registered DNS resolver. This allows the test to
+// mock the DNS resolution for the proxy resolver. It also registers the
+// original DNS resolver after the test is done.
+func setupDNS(t *testing.T) *manual.Resolver {
+	mr := manual.NewBuilderWithScheme("dns")
+
+	dnsResolverBuilder := resolver.Get("dns")
+	resolver.Register(mr)
+
+	t.Cleanup(func() { resolver.Register(dnsResolverBuilder) })
+	return mr
+}
+
+// proxyAddressWithTargetAttribute creates a resolver.Address for the proxy,
+// adding the target address as an attribute.
+func proxyAddressWithTargetAttribute(proxyAddr string, targetAddr string) resolver.Address {
+	addr := resolver.Address{Addr: proxyAddr}
+	addr = proxyattributes.Populate(addr, proxyattributes.Options{
+		ConnectAddr: targetAddr,
+	})
+	return addr
+}
+
+// Tests the scenario where proxy is configured and the target URI contains the
+// "dns" scheme and target resolution is enabled. The test verifies that the
+// addresses returned by the delegating resolver combines the addresses
+// returned by the proxy resolver and the target resolver.
+func (s) TestDelegatingResolverwithDNSAndProxyWithTargetResolution(t *testing.T) {
+	const (
+		targetTestAddr          = "test.com"
+		resolvedTargetTestAddr1 = "1.2.3.4:8080"
+		resolvedTargetTestAddr2 = "1.2.3.5:8080"
```

Comment:
These address strings are too close to each other, and would be hard to differentiate when looking at a failing test's logs. I would suggest something like:

	const (
		targetTestAddr          = "test.com"
		resolvedTargetTestAddr1 = "1.1.1.1:8080"
		resolvedTargetTestAddr2 = "2.2.2.2:8080"
		envProxyAddr            = "proxytest.com"
		resolvedProxyTestAddr1  = "11.11.11.11:7687"
		resolvedProxyTestAddr2  = "22.22.22.22:7687"
	)